packages/sit-onyx/src/components/OnyxSelect/types.ts

```diff
@@ -0,0 +1,56 @@
+import type { RequiredMarkerProp } from "../../composables/required";
+
+export const MULTISELECT_TEXT_MODE = ["summary", "preview"] as const;
+export type MultiselectTextMode = (typeof MULTISELECT_TEXT_MODE)[number];
+
+export type OnyxSelectProps<TValue extends string | string[]> = RequiredMarkerProp & {
+  /**
+   * Current value of the select.
+   * TODO: change the type after the flyout gets added and the select becomes a real interactive component!
+   */
+  modelValue?: TValue;
+  /**
+   * Label to show above the select. Required due to accessibility / screen readers.
+   * If you want to visually hide the label, use the `hideLabel` property.
+   */
+  label: string;
+  /**
+   * If `true`, the label will be visually hidden and the `title` attribute will be set.
+   * For accessibility / screen readers, the aria-label will still be set.
+   */
+  hideLabel?: boolean;
+  /**
+   * Whether the select should be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Whether to show a skeleton select.
+   * TODO: implement skeleton
+   */
+  skeleton?: boolean;
+  /**
+   * Shows a loading indicator.
+   */
+  loading?: boolean;
+  /**
+   * Placeholder to show when the value is empty.
+   */
+  placeholder?: string;
+  /**
+   * Whether multiple values can be selected.
+   * TODO: clarify, if "multiple" should be derived implicitly by TValue?
```

Comment:
Do you mean that if TValue is not an array, we have to convert it to be an array?
Or restrict the allowed typings to be array?

The alternative Idea was to make "multiple" not a boolean prop but a internal computed that checks whether TValue is an array. the only situation this would not work is if modelValue is initialized undefined instead of an empty array. then it would assume that a single select is desired.

What do you think?